plugins/module_utils/dependency_checker.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) IBM Corporation 2025
+# Licensed under the Apache License, Version 2.0
+
+from __future__ import absolute_import, division, print_function
+import re
+import subprocess
+import sys
+import json
+
+__metaclass__ = type
+
+# ------------------------------
+# Compatibility Matrix
+# ------------------------------
+COMPATIBILITY_MATRIX = [
+    {"zoau_version": "1.3.5", "python_version": "3.12", "galaxy_core_version": "1.12.0", "zos_range": "2.5-3.1"},
+    {"zoau_version": "1.3.5.1", "python_version": "3.10", "galaxy_core_version": "1.12.0", "zos_range": "2.5-3.1"},
+    {"zoau_version": "1.3.6.0", "python_version": "3.10", "galaxy_core_version": "1.10.1", "zos_range": "2.5-3.1"},
+    {"zoau_version": "1.4.0", "python_version": "3.11", "galaxy_core_version": "1.11.0", "zos_range": "2.5-3.1"},
+    {"zoau_version": "1.4.0", "python_version": "3.11", "galaxy_core_version": "1.12.0", "zos_range": "2.5-3.1"},
+]
+
+# ------------------------------
+# Helpers
+# ------------------------------
+def run_command(module, cmd):
+    rc, out, err = module.run_command(cmd)
+    if rc != 0:
+        module.fail_json(msg=f"Command failed: {cmd}", stdout=out, stderr=err)
+    return out.strip()
+
+# ------------------------------
+# Version Fetchers
+# ------------------------------
+def get_zoau_version(module):
+    output = run_command(module, "zoaversion")
```

Comment:
No need to run `zoaversion` command, as we spoke before better to use the `zoau_dependency_checker` functionality.

```python
try:
    from zoautil_py import ZOAU_API_VERSION
except Exception:
    ZOAU_API_VERSION = "1.4.0"
```